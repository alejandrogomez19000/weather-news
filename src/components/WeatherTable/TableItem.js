import React from "react";
import "./WeatherTable.scss";
import propTypes from "prop-types";

const TableItem = ({
  maxTemp,
  minTemp,
  day,
  date,
  description,
  humidity,
  icon,
}) => (
  <div className="table-item">
    <div className="date-container">
      <span>{day}</span>
      <span className="date">{date}</span>
    </div>
    <img src={icon} className="table-item-icon" alt="weather" />
    <div className="temp-container">
      <span className="max-temp">{maxTemp}Â°</span>
      <span>/{minTemp}</span>
    </div>
    <div className="description">{description}</div>
    <div className="humidity">
      <span>Hum.</span>
      {humidity}%
    </div>
  </div>
);

TableItem.propTypes = {
  maxTemp: propTypes.number.isRequired,
  minTemp: propTypes.number.isRequired,
  day: propTypes.string.isRequired,
  date: propTypes.string.isRequired,
  description: propTypes.string.isRequired,
  humidity: propTypes.number.isRequired,
  icon: propTypes.string.isRequired,
};

export default TableItem;
