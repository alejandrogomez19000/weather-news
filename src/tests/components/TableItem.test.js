import React from "react";
import TableItem from "components/WeatherTable/TableItem";
import ShallowRender from "react-test-renderer/shallow";

const renderer = new ShallowRender();

const tableProps = {
  maxTemp: 20,
  minTemp: 15,
  day: "some day",
  date: "some date",
  description: "some description",
  humidity: 80,
  icon: "some icon",
};

describe("render component", () => {
  it("should be render", async () => {
    renderer.render(<TableItem {...tableProps} />);
    const renderOutput = renderer.getRenderOutput();
    expect(renderOutput).toMatchSnapshot();
  });
});
