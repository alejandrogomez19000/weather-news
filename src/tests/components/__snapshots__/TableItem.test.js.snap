// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component should be render 1`] = `
<div
  className="table-item"
>
  <div
    className="date-container"
  >
    <span>
      some day
    </span>
    <span
      className="date"
    >
      some date
    </span>
  </div>
  <img
    alt="weather"
    className="table-item-icon"
    src="some icon"
  />
  <div
    className="temp-container"
  >
    <span
      className="max-temp"
    >
      20
      Â°
    </span>
    <span>
      /
      15
    </span>
  </div>
  <div
    className="description"
  >
    some description
  </div>
  <div
    className="humidity"
  >
    <span>
      Hum.
    </span>
    80
    %
  </div>
</div>
`;
